canfdtest.c

```diff
@@ -123,17 +123,40 @@ static void millisleep(int msecs)
 	if (msecs <= 0) {
 		sched_yield();
 	} else {
-		struct timespec rqtp, rmtp;
-
-		/* sleep in ms */
-		rqtp.tv_sec = msecs / 1000;
-		rqtp.tv_nsec = (msecs % 1000) * 1000000;
-		while (nanosleep(&rqtp, &rmtp)) {
-			if (errno != EINTR) {
-				printf("t\n");
-				break;
+		struct timespec curtp;
+		int err;
+
+		err = clock_gettime(CLOCK_MONOTONIC, &curtp);
```

Comment:
This looks quite complicated, why don't you use a `clock_nanosleep` with a `CLOCK_MONOTONIC` and do a relative sleep.